src/sentry/scim/endpoints/users.py

```diff
@@ -0,0 +1,211 @@
+from django.conf import settings
+from django.db import transaction
+from django.db.models import Q
+from rest_framework import serializers
+from rest_framework.response import Response
+
+from sentry import roles
+from sentry.api.endpoints.organization_member_details import OrganizationMemberDetailsEndpoint
+from sentry.api.endpoints.organization_member_index import OrganizationMemberSerializer
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.paginator import GenericOffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.organization_member import OrganizationMemberSCIMSerializer
+from sentry.api.validators import AllowedEmailField
+from sentry.models import AuditLogEntryEvent, AuthIdentity, OrganizationMember
+from sentry.signals import member_invited
+
+from .utils import SCIM_API_ERROR, SCIMEndpoint, parse_filter_conditions
+
+ERR_ONLY_OWNER = "You cannot remove the only remaining owner of the organization."
+
+SCIM_404_USER_RES = {
+    "schemas": [SCIM_API_ERROR],
+    "detail": "User not found.",
+}
+
+SCIM_409_USER_EXISTS = {
+    "schemas": ["urn:ietf:params:scim:api:messages:2.0:Error"],
+    "detail": "User already exists in the database.",
+}
+
+
+class SCIMUserSerializer(serializers.Serializer):
+    email = AllowedEmailField(max_length=75, required=True)
+    role = serializers.ChoiceField(choices=roles.get_choices(), required=True)
+    sendInvite = serializers.BooleanField(required=False, default=True, write_only=True)
+
+    def validate_email(self, email):
+        queryset = OrganizationMember.objects.filter(
+            Q(email=email) | Q(user__email__iexact=email, user__is_active=True),
+            organization=self.context["organization"],
+        )
+
+        if queryset.filter.exists():
+            raise serializers.ValidationError("User already exists in the database.")
+        return email
+
+
+class OrganizationSCIMUserDetails(SCIMEndpoint, OrganizationMemberDetailsEndpoint):
+    def _get_member(self, organization, member_id):
+        try:
+            member = OrganizationMember.objects.get(organization=organization, id=member_id)
+        except OrganizationMember.DoesNotExist:
+            raise ResourceDoesNotExist
+        except AssertionError as error:
+            if str(error) == "value too large":
+                raise ResourceDoesNotExist
+            raise error
```

Comment:
You could handle validating the member_id in a serializer so that you're not sending bad data to the database and hitting these assertion errors.